File: works/tmp/13015.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;
import java.util.Collections;

public class Main
{
    public static void main(String[] args) throws IOException
    {
        Scanner sc = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();

        int N = sc.nextInt();
        int w = 4 * N - 3;
        int h = 2 * N - 1;
        int mid = w-N*2; // 가운데 공백
        ArrayList<ArrayList<Character>> board = new ArrayList<ArrayList<Character>>();

        // 첫 줄
        ArrayList<Character> a = new ArrayList<>();
        for(int i = 0; i < N; i++) a.add('*');
        for(int i = 0; i < mid; i++) a.add(' ');
        for(int i = 0; i < N; i++) a.add('*');
        board.add(a);
        mid-=2;

        int l=0;
        for(int i = 1; i < N; i++)
        {
            a = new ArrayList<>();
            for(int j = 0; j < i; j++) a.add(' ');
            a.add('*');
            for(int j = 0; j < N-2; j++) a.add(' ');
            a.add('*');
            for(int j = 0; j < mid; j++) a.add(' ');
            if(mid > 0) a.add('*');
            mid-=2;
            for(int j = 0; j < N-2; j++) a.add(' ');
            a.add('*');
            l = i-1;
            board.add(a);
        }
        mid = 1;

        for(int i = N; i < h-1; i++)
        {
            a = new ArrayList<>();
            for(int j = 0; j < l; j++) a.add(' ');
            l--;
            a.add('*');
            for(int j = 0; j < N-2; j++) a.add(' ');
            a.add('*');
            for(int j = 0; j < mid; j++) a.add(' ');
            mid += 2;
            a.add('*');
            for(int j = 0; j <N-2; j++) a.add(' ');
            a.add('*');
            board.add(a);
        }

        a = new ArrayList<>();
        for(int i = 0; i < N; i++) a.add('*');
        for(int i = 0; i < mid; i++) a.add(' ');
        for(int i = 0; i <N; i++) a.add('*');
        board.add(a);

        for(int i = 0; i < h; i++)
        {
            for(Character x : board.get(i)) System.out.print(x);
            System.out.println();
        }
    }
}

File: works/tmp/13015.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.nio.Buffer
import java.util.*


fun main()
{
    val sc = Scanner(System.`in`)
    val N = sc.nextInt()
    val w = 4 * N - 3
    val h = 2 * N - 1
    var mid = w - N * 2
    var board = ArrayList<ArrayList<Char>>()

    //
    var a = ArrayList<Char>()
    for(i in 0 until N) a.add('*')
    for(i in 0 until mid) a.add(' ')
    for(i in 0 until N) a.add('*')
    board.add(a)
    mid-=2

    var l = 0
    for(i in 1 until N)
    {
        a = ArrayList<Char>()
        for(j in 0 until i) a.add(' ')
        a.add('*')
        for(j in 0 until N-2) a.add(' ')
        a.add('*')
        for(j in 0 until mid) a.add(' ')
        if(mid > 0) a.add('*')
        mid-=2
        for(j in 0 until N-2) a.add(' ')
        a.add('*')
        l = i-1
        board.add(a)
    }
    mid = 1

    for(i in N until h-1)
    {
        a = ArrayList<Char>()
        for(j in 0 until l) a.add(' ')
        l--
        a.add('*')
        for(j in 0 until N-2)  a.add(' ')
        a.add('*')
        for(j in 0 until mid) a.add(' ')
        mid += 2
        a.add('*')
        for(j in 0 until N-2) a.add(' ')
        a.add('*')
        board.add(a)
    }

    a = ArrayList<Char>()
    for(i in 0 until N) a.add('*')
    for(i in 0 until mid) a.add(' ')
    for(i in 0 until N) a.add('*')
    board.add(a)

    for(i in 0 until h)
    {
        for(x in board[i]) print(x)
        println()
    }

}